pkg/hubble/metrics/port-distribution/handler_test.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package portdistribution
+
+import (
+	"context"
+	"testing"
+
+	pb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/hubble/metrics/api"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPortDistributionHandler(t *testing.T) {
+	registry := prometheus.NewRegistry()
+	opts := api.Options{"sourceContext": "namespace", "destinationContext": "namespace"}
+
+	portHandler := &portDistributionHandler{}
+
+	t.Run("Init", func(t *testing.T) {
+		require.NoError(t, portHandler.Init(registry, opts))
+	})
+
+	t.Run("Status", func(t *testing.T) {
+		require.Equal(t, "destination=namespace,source=namespace", portHandler.Status())
+	})
+
+	t.Run("ProcessFlow_SkipReply", func(t *testing.T) {
+		flow := &pb.Flow{
+			EventType: &pb.CiliumEventType{Type: monitorAPI.MessageTypePolicyVerdict},
+			L4: &pb.Layer4{
+				Protocol: &pb.Layer4_TCP{
+					TCP: &pb.TCP{
+						DestinationPort: 8080,
+					},
+				},
+			},
+			Source:      &pb.Endpoint{Namespace: "foo"},
+			Destination: &pb.Endpoint{Namespace: "bar"},
+			Verdict:     pb.Verdict_FORWARDED,
+			IsReply:     &wrappers.BoolValue{Value: true},
+		}
+		portHandler.ProcessFlow(context.TODO(), flow)
+
+		metricFamilies, err := registry.Gather()
+		require.NoError(t, err)
+
+		require.Empty(t, metricFamilies)
+	})
+
+	t.Run("ProcessFlow_SkipDropped", func(t *testing.T) {
+		flow := &pb.Flow{
+			EventType: &pb.CiliumEventType{Type: monitorAPI.MessageTypePolicyVerdict},
+			L4: &pb.Layer4{
+				Protocol: &pb.Layer4_TCP{
+					TCP: &pb.TCP{
+						DestinationPort: 8080,
+					},
+				},
+			},
+			Source:      &pb.Endpoint{Namespace: "foo"},
+			Destination: &pb.Endpoint{Namespace: "bar"},
+			Verdict:     pb.Verdict_DROPPED,
+			IsReply:     &wrappers.BoolValue{Value: false},
+		}
+		portHandler.ProcessFlow(context.TODO(), flow)
+
+		metricFamilies, err := registry.Gather()
+		require.NoError(t, err)
+
+		require.Empty(t, metricFamilies)
+	})
+
+	t.Run("ProcessFlow", func(t *testing.T) {
+		flow := &pb.Flow{
```

Comment:
It might be a good idea to process multiple flows, (some with different ports, and some with the same) and assert the metric counter is at the right value for different ports.

Reply:
I added a new test case.